src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -128,7 +128,15 @@ NewTerminalArgs Pane::GetTerminalArgsForPane() const
     auto controlSettings = _control.Settings().as<TerminalSettings>();
 
     args.Profile(controlSettings.ProfileName());
-    args.StartingDirectory(controlSettings.StartingDirectory());
+    // If we know the user's working directory use it instead of the profile.
+    if (!_control.WorkingDirectory().empty())
+    {
+        args.StartingDirectory(_control.WorkingDirectory());
+    }

```

Comment:
You can prevent the double-COM-call with
```suggestion
    if (const auto path = _control.WorkingDirectory(); !path.empty())
    {
        args.StartingDirectory(path);
    }
```

I love the syntax... it strongly reminds me of Go.